src/main/game/GameModel.java

```diff
@@ -23,26 +34,50 @@
     private boolean gameHasStarted = false;
     
     public void initializeGame() {
-        this.players = new ArrayList<Player>();
-        Player human = new Player(0, HUMAN_STARTING_LOCATION);
-        Player panda = new Player(1, PANDA_STARTING_LOCATION);
-        currentPlayer = human;
-        players.add(human);
-        players.add(panda);
-        gBoard = new Gameboard(players);
-        human.initializeSimpleTiles(gBoard.getTiles());
-        panda.initializeSimpleTiles(gBoard.getTiles());
-        human.updateSimpleTiles(gBoard.getTiles());
-        panda.updateSimpleTiles(gBoard.getTiles());
-        gameHasStarted =  true;
-        startTurn();
+        try {
+            this.players = new ArrayList<Player>();
+            Player human = new Player(0, HUMAN_STARTING_LOCATION);
+            Player panda = new Player(1, PANDA_STARTING_LOCATION);
+            currentPlayer = human;
+            players.add(human);
+            players.add(panda);
+            gBoard = new Gameboard(players);
+            initialUnit(human, panda);
+            human.initializeSimpleTiles(gBoard.getTiles());
+            panda.initializeSimpleTiles(gBoard.getTiles());
+            human.updateSimpleTiles(gBoard.getTiles());
+            panda.updateSimpleTiles(gBoard.getTiles());
+            gameHasStarted = true;
+            startTurn();
+        }catch(GameFailedToStartException e){
+            System.out.println(e.getMessage());
+        }
+
     }
     
     public void updateGame() { //This is called up to 60 times per second
     	if (gameHasStarted) {
     		checkIfGameOver();
     	}
     }
+
+    public void initialUnit(Player human, Player panda) throws GameFailedToStartException {
```

Comment:
can you change this to initializeUnits? or addInitialUnitsToGameboard? something like that